systemtest/benchtest/expvar.go

```diff
@@ -98,3 +144,57 @@ func WaitUntilServerInactive(ctx context.Context, server string) error {
 	}
 	return nil
 }
+
+// NOTE(marclop): There are some fields that aren't being aggregated
+// like the circular buffers PauseNS, PauseEnd, BySize, and booleans
+// (DebugGC, EnableGC).
+func aggregateMemStats(from runtime.MemStats, to *runtime.MemStats) {
+	to.Alloc += from.Alloc
+	to.TotalAlloc += from.TotalAlloc
+	to.Sys += from.Sys
+	to.Lookups += from.Lookups
+	to.Mallocs += from.Mallocs
+	to.Frees += from.Frees
+	to.HeapAlloc += from.HeapAlloc
+	to.HeapSys += from.HeapSys
+	to.HeapIdle += from.HeapIdle
+	to.HeapInuse += from.HeapInuse
+	to.HeapReleased += from.HeapReleased
+	to.HeapObjects += from.HeapObjects
+	to.StackInuse += from.StackInuse
+	to.StackSys += from.StackSys
+	to.MSpanInuse += from.MSpanInuse
+	to.MSpanSys += from.MSpanSys
+	to.MCacheInuse += from.MCacheInuse
+	to.MCacheSys += from.MCacheSys
+	to.BuckHashSys += from.BuckHashSys
+	to.GCSys += from.GCSys
+	to.OtherSys += from.OtherSys
+	to.NextGC += from.NextGC
+	to.LastGC += from.LastGC
+	to.PauseTotalNs += from.PauseTotalNs
+	to.NumGC += from.NumGC
+	to.NumForcedGC += from.NumForcedGC
+	to.GCCPUFraction += from.GCCPUFraction
+}
+
+func aggregateLibbeatStats(from LibbeatStats, to *LibbeatStats) {
+	to.ActiveEvents += from.ActiveEvents
```

Comment:
@marclop (Apologies for putting the message after the PR is reviewed and merged)

IIUC, some of these metrices are monotonic counters (strictly increasing for the lifetime of the apm-server process). In this case since we are aggregating different servers, it might result in a case where taking a diff between 2 timestamps (t0, t1 where t1 > t0) will result in an unexpected negative value if query over one expvar fails or one apm-server just dies. Is this correct? Do we need to take this into account?

Reply:
> t might result in a case where taking a diff between 2 timestamps (t0, t1 where t1 > t0) will result in an unexpected negative value if query over one expvar fails or one apm-server just dies.

as you mention, the negative result would happen if an APM Server has been restarted (due to running out of memory). We could implement some protection against this by writing more logic that protects against this case. We can open an improvement issue to change the way this works to account for this case.

I think you mean to say where 2 timestamps (t0, t1 where t0 > t1), not the opposite, right?

Reply:
I meant `t1 - t0`

Reply:
Makes sense, we can add some protection to protect us from these cases. BTW, I think in the future we will have to look at a different way to get these metrices (we can discuss this later).